third-party/realdds/src/topics/notification-msg.cpp

```diff
@@ -0,0 +1,29 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/topics/notification/notification-msg.h>
+#include <realdds/topics/notification/notificationPubSubTypes.h>
+
+#include <realdds/dds-topic.h>
+#include <realdds/dds-utilities.h>
+
+#include <fastdds/dds/topic/Topic.hpp>
+
+
+namespace realdds {
+namespace topics {
+namespace device {
+
+
+/*static*/ std::shared_ptr< dds_topic >
+notification::create_topic( std::shared_ptr< dds_participant > const & participant, char const * topic_name )
```

Comment:
I think it is better to pass `topic_root_name` and let this function build the full name, for client/server consistency.

Reply:
But this is a general-purpose function, not one that requires a hierarchy -- you can create a notification topic anywhere you please. It's the device that requires a hierarchy, therefore the "building the full name" needs to live there.